frontend/src/components/Permissions.tsx

```diff
@@ -107,213 +157,240 @@ const Permissions: React.FC<PermissionsProps> = ({
+                      }}
+                    />
                   </div>
                 )}
-              </li>
-            );
-          })}
-        </ul>
-      </div>
+              </>
+            )}
+          </>
+        )}
 
-      {(
-        isNewConnection ? !permissions.expiresAt || days : canEditPermissions
-      ) ? (
-        <>
-          {!expireOptions && (
-            <Button
-              type="button"
-              variant="secondary"
-              onClick={() => setExpireOptions(true)}
-            >
-              <PlusCircle className="w-4 h-4 mr-2" />
-              Set expiration date
-            </Button>
-          )}
+      {!expireOption && (
+        <Button
+          type="button"
+          variant="secondary"
+          onClick={() => setExpireOption(true)}
+          className="mb-6"
+        >
+          <PlusCircle className="w-4 h-4 mr-2" />
+          Set expiration time
+        </Button>
+      )}
 
-          {expireOptions && (
-            <div className="mt-5">
-              <p className="font-medium text-sm mb-2">Connection expiration</p>
-              {!isNewConnection && (
-                <p className="mb-2 text-muted-foreground text-sm">
-                  Expires:{" "}
-                  {permissions.expiresAt &&
-                  new Date(permissions.expiresAt).getFullYear() !== 1
-                    ? new Date(permissions.expiresAt).toString()
-                    : "This app will never expire"}
-                </p>
-              )}
-              <div id="expiry-days" className="grid grid-cols-4 gap-2 text-xs">
-                {Object.keys(expiryOptions).map((expiry) => {
-                  return (
-                    <div
-                      key={expiry}
-                      onClick={() => handleDaysChange(expiryOptions[expiry])}
-                      className={cn(
-                        "cursor-pointer rounded border-2 text-center py-4",
-                        days == expiryOptions[expiry]
-                          ? "border-primary"
-                          : "border-muted"
-                      )}
-                    >
-                      {expiry}
-                    </div>
-                  );
-                })}
-              </div>
-            </div>
-          )}
-        </>
-      ) : (
-        <>
-          <p className="text-sm font-medium mb-2">Connection expiry</p>
-          <p className="text-muted-foreground text-sm">
-            {permissions.expiresAt &&
-            new Date(permissions.expiresAt).getFullYear() !== 1
-              ? new Date(permissions.expiresAt).toString()
-              : "This app will never expire"}
-          </p>
-        </>
+      {expireOption && (
+        <div className="mb-6">
+          <p className="font-medium text-sm mb-2">Connection expiration</p>
+          <div className="grid grid-cols-2 md:grid-cols-6 gap-2 text-xs mb-4">
+            {Object.keys(expiryOptions).map((expiry) => {
```

Comment:
if you pass an `expires_at` it should not be editable and there should not be options to choose from

Reply:
After writing the logic, I felt maybe editable is not necessary(?)

We show the user what all options have been marked and the user can simply click next. If they want to change anything they can. Instead of adding more logic we can maybe add a alert box on top which says "The app might not function properly without the following permissions, change them at your own risk" (probably something less threatening)